pubtools/_quay/quay_client.py

```diff
@@ -32,52 +37,55 @@ def __init__(self, username, password, host=None):
         self.password = password
         self.session = QuaySession(hostname=host, api="docker")
 
-    def get_manifest_list(self, image, raw=False):
+    def get_manifest(self, image, raw=False, manifest_list=False):
         """
-        Get manifest list of a given image. Throw an error if it couldn't be retrieved.
+        Get manifest of a given image along with its type.
+
+        If reference contains multiple manifests, manifest list will be returned.
 
         Args:
             image (str):
                 Image for which to get the manifest list.
             raw (bool):
                 Whether to return the manifest as raw JSON.
+            manifest_list (bool):
+                Whether to only return a manifest list and raise an exception otherwise.
         Returns (dict|str):
-            The returned manifest list.
-        Raises:
+            Image manifest
+                Raises:
             ManifestTypeError:
                 When the image doesn't have a manifest list.
         """
-        LOG.info("Getting manifest list of image {0}".format(image))
         repo, ref = self._parse_and_validate_image_url(image)
         endpoint = "{0}/manifests/{1}".format(repo, ref)
+
         # request 'Content-Type' to be manifest list
         kwargs = {"headers": {"Accept": QuayClient.MANIFEST_LIST_TYPE}}
-
         response = self._request_quay("GET", endpoint, kwargs)
-        if response.headers["Content-Type"] != QuayClient.MANIFEST_LIST_TYPE:
+        if manifest_list and response.headers["Content-Type"] != QuayClient.MANIFEST_LIST_TYPE:
             raise ManifestTypeError("Image {0} doesn't have a manifest list".format(image))
 
         if raw:
             return response.text
         else:
             return response.json()
 
-    def upload_manifest_list(self, manifest_list, image):
+    def upload_manifest(self, manifest, image):
         """
-        Upload a manifest list to a specified image.
+        Upload manifest to a specified image.
```

Comment:
To avoid confusion, might be useful to add a comment here specifying that manifest/manifest_list can be uploaded.

Reply:
Added.